examples/seesaw_ano_rotary_7segment_demo.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-FileCopyrightText: 2021 John Furcean
+# SPDX-License-Identifier: MIT
+
+"""I2C ANO rotary encoder with 7 segment display example."""
+
+import board
+from adafruit_ht16k33 import segments
+from adafruit_seesaw import seesaw, rotaryio, digitalio
+
+# For use with the STEMMA connector on QT Py RP2040
+# import busio
+# i2c = busio.I2C(board.SCL1, board.SDA1)
+# seesaw = seesaw.Seesaw(i2c, 0x49)
+
+i2c = board.I2C()  # uses board.SCL and board.SDA
+# i2c = board.STEMMA_I2C()  # For using the built-in STEMMA QT connector on a microcontroller
+seesaw = seesaw.Seesaw(i2c, addr=0x49)
+display = segments.Seg14x4(i2c, address=0x70)
+
+seesaw_product = (seesaw.get_version() >> 16) & 0xFFFF
+print("Found product {}".format(seesaw_product))
+if seesaw_product != 5740:
+    print("Wrong firmware loaded?  Expected 5740")
+
+seesaw.pin_mode(1, seesaw.INPUT_PULLUP)
+seesaw.pin_mode(2, seesaw.INPUT_PULLUP)
+seesaw.pin_mode(3, seesaw.INPUT_PULLUP)
+seesaw.pin_mode(4, seesaw.INPUT_PULLUP)
+seesaw.pin_mode(5, seesaw.INPUT_PULLUP)
+
+select = digitalio.DigitalIO(seesaw, 1)
+select_held = False
+up = digitalio.DigitalIO(seesaw, 2)
+up_held = False
+left = digitalio.DigitalIO(seesaw, 3)
+left_held = False
+down = digitalio.DigitalIO(seesaw, 4)
+down_held = False
+right = digitalio.DigitalIO(seesaw, 5)
+right_held = False
+
+encoder = rotaryio.IncrementalEncoder(seesaw)
+last_position = None
+
+buttons = [select, up, left, down, right]
+button_names = ["Select", "Up", "Left", "Down", "Right"]
+button_states = [select_held, up_held, left_held, down_held, right_held]
+seven_segment_names = ["SELE", " UP ", "LEFT", "DOWN", "RIGH"]
+
+while True:
+    position = encoder.position
+
+    if position != last_position:
+        last_position = position
+        display.print("   %s" % str(position))
+        print("Position: {}".format(position))
+    for b in range(5):
```

Comment:
Might be more complex for beginners, but this is an excellent use case for `zip()`:

```python
    for button, state, button_name, segment_name in zip(buttons, button_states, button_names, seven_segment_names):
        ...
```

Reply:
ohh cool! i did not know about `zip()`. i tried this:

```
for button, state, button_name, segment_name in zip(buttons, button_states, button_names, seven_segment_names):
        if not button.value and state is False:
            state = True
            display.print(segment_name)
            print("{} button pressed".format(button_name))
        if button.value and state is True:
            state = False
            display.print("    ")
            print("{} button released".format(button_name))
```
and the only thing is it doesn't seem to be reading the release and its continuously reading a press while a button is held down whereas before it would just read it once. not sure if there's something else i should be adding with `zip()`?

Reply:
Ah, I think it's because the object `state` equates to isn't actually updated in this case, since you're not updating an item in a list, just setting the value of the `state` variable.  `enumerate()` to the rescue!

```python
    for index, (button, state, button_name, segment_name) in zip(buttons, button_states, button_names, seven_segment_names):
        ...
```

I do admit at this point your original code is probably simpler and more elegant haha

Reply:
ohh i see. yeah i think for these demos `enumerate(zip())` might be getting a little too in the weeds since that index is needed but i will definitely keep that in mind for project code, super handy! i'll update the PR with the f-strings.